python_ta/reporters/core.py

```diff
@@ -263,7 +270,7 @@ def on_close(self, stats, previous_stats):
 
         Close the reporter's output stream (if not sys.stdout).
         """
-        if self.out is not sys.stdout:
+        if self.out is not sys.stdout and not self.filepath_io:
```

Comment:
Now that you've introduced the new attribute, remove the check for `self.is is not sys.stdout`. Instead, in the `set_output` method above set the attribute appropriately when `None` is passed in.